internal/app/app.go

```diff
@@ -145,8 +147,19 @@ func NewApp(ctx context.Context, cfg *config.Config, logger *zap.SugaredLogger)
 		return nil, err
 	}
 
+	// get matrix hostname without schema
+	matrixHostname := cfg.Matrix.Host
+	if !strings.Contains(matrixHostname, "://") {
+		matrixHostname = "dummy://" + matrixHostname
```

Comment:
This part can get confusing after we forgot why we did it. Maybe we should add a comment explaining why?
```suggestion
		// Add dummy protocol to make the url pkg happy, 
		// we just want to extract the hostname
		matrixHostname = "dummy://" + matrixHostname
```